rts/Lua/LuaSyncedCtrl.cpp

```diff
@@ -1424,6 +1428,70 @@ int LuaSyncedCtrl::ShareTeamResource(lua_State* L)
 	return 0;
 }
 
+/*** Adjusts a team's resource statistics, for post-game graphs.
+ *
+ * @function Spring.AddTeamResourceStats
+ * @param teamID integer
+ * @param amounts table
+ * @return nil
+ */
+int LuaSyncedCtrl::AddTeamResourceStats(lua_State* L)
+{
+	const auto team = ParseTeam(L, __func__, 1);
+	if (team == nullptr)
+		return 0;
+
+	if (!lua_istable(L, 2))
+		return 0;
+
+	auto& stats = team->GetCurrentStats();
+	for (lua_pushnil(L); lua_next(L, 2) != 0; lua_pop(L, 1)) {
+		if (!lua_isstring(L, LUA_TABLE_KEY_INDEX))
+			continue;
+		if (!lua_istable(L, LUA_TABLE_VALUE_INDEX))
+			continue;
+
+		using TS = TeamStatistics;
+		std::array <float TS::*, SResourcePack::MAX_RESOURCES> entries;
+		switch (hashString(lua_tostring(L, LUA_TABLE_KEY_INDEX))) {
+			case hashString("used")    : entries = {&TS::metalUsed    , &TS::energyUsed    }; break;
+			case hashString("produced"): entries = {&TS::metalProduced, &TS::energyProduced}; break;
+			case hashString("excess")  : entries = {&TS::metalExcess  , &TS::energyExcess  }; break;
+			case hashString("received"): entries = {&TS::metalReceived, &TS::energyReceived}; break;
+			case hashString("sent")    : entries = {&TS::metalSent    , &TS::energySent    }; break;
+			default: continue;
+		}
+
+		for (int i = 0; i < entries.size(); ++i) {
+			// Parse by resource index, like `produced = { 1.23, 45.6 }`
```

Comment:
The inside of this loop should probably end up as some sort of helper utility function eventually.